frontend/src/lib/dashbaordVariables/getDashboardVariables.ts

```diff
@@ -1,33 +1,29 @@
 import getStartEndRangeTime from 'lib/getStartEndRangeTime';
 import store from 'store';
+import { Dashboard } from 'types/api/dashboard/getAll';
 
-export const getDashboardVariables = (): Record<string, unknown> => {
-	try {
-		const {
-			globalTime,
-			dashboards: { dashboards },
-		} = store.getState();
-		if (dashboards.length > 0) {
-			const [selectedDashboard] = dashboards || [];
-			const {
-				data: { variables = {} },
-			} = selectedDashboard;
+export const getDashboardVariables = (
+	variables?: Dashboard['data']['variables'],
+): Record<string, unknown> => {
+	if (!variables) {
+		return {};
+	}
 
-			const { start, end } = getStartEndRangeTime({
-				type: 'GLOBAL_TIME',
-				interval: globalTime.selectedTime,
-			});
+	try {
+		const { globalTime } = store.getState();
```

Comment:
We should move this to context

Reply:
different reducer in future we do have plan for same